exonum-java-binding/core/rust/src/storage/proof_list_index.rs

```diff
@@ -369,6 +369,19 @@ pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofListInd
     handle::drop_handle::<ProofListIndexIter<Value>>(&env, iter_handle);
 }
 
+fn make_java_proof_root<'a>(
+    env: &JNIEnv<'a>,
+    proof: &ListProof<Value>,
```

Comment:
@vitvakatu  Why doesn't ListProof in Rust contain the length?

Reply:
I have no idea https://github.com/exonum/exonum/blob/master/components/merkledb/src/proof_list_index/proof.rs#L144

Reply:
That is really strange — why force a user provide the size when we assume they don't know _anything_ about the list (including its size!), don't trust the system, hence need a proof. How do they know the size (without trusting the system)? If they do trust it, then the proof is not needed at all.

Reply:
Yep, looks like it violates our privacy guarantees. Created https://jira.bf.local/browse/ECR-3414

Reply:
Maybe I missing something, but in rust, we force the user to provide list length only when we validating the proof. We don't need list size to construct the proof.

Reply:
> we force the user to provide list length only when we validating the proof.

Why does the core exclude the essential part of the proof? The proof cannot be verified without this piece of data. Why is the user forced to obtain that missing piece of the proof manually?